src/App.tsx

```diff
@@ -14,11 +14,118 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[];
+  clickStart: boolean;
+  visibleGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    clickStart: false,
+    visibleGoods: [...goodsFromServer],
+  };
+
+  getReset() {
+    this.setState(prevState => ({ visibleGoods: [...prevState.goods] }));
+  }
+
+  getReverse() {
+    this.setState((prevState) => (
+      { visibleGoods: prevState.visibleGoods.reverse() }
```

Comment:
Reverse method mutates the initial array, you should make a copy before mutating